test/e2e_tests/pageManager/webapp/pages/conversation.page.ts

```diff
@@ -116,6 +132,27 @@ export class ConversationPage {
     return false;
   }
 
+  async isSystemMessageVisible(messageText: string) {
+    // Trying multiple times for the message to appear
+    for (let i = 0; i < 10; i++) {
+      // Wait for at least one matching element to appear (optional timeout can be set)
+      await this.systemMessages.first().waitFor({state: 'visible'});
+
+      // Then get all matching elements
+      const messages = await this.systemMessages.all();
+      for (const message of messages) {
+        const messageTextContent = await message.textContent();
+        if (messageTextContent?.includes(messageText)) {
+          continue;
+        }
+        return true;
+      }
+      await this.page.waitForTimeout(500); // Wait for 0.5 second before next attempt
+    }
+
+    return false;
+  }
+
```

Comment:
why is it necessary to try 10 times? shouldn't waitFor: visible wait until message appears?